src/transformers/models/bart/modeling_bart.py

```diff
@@ -257,7 +260,13 @@ def forward(
 
         attn_output = attn_output.view(bsz, self.num_heads, tgt_len, self.head_dim)
         attn_output = attn_output.transpose(1, 2)
-        attn_output = attn_output.reshape(bsz, tgt_len, embed_dim)
+
+        # Use the embed_dim from class rather than hidden_state, this is due to
+        # the reason that attn_output can be partitioned across GPUs
+        # when using tensor-parallelism, in which case the embed_dimension from
+        # the input is not equal to the attention's last dimension after merging
+        # heads.
```

Comment:
We have a 119 char limits so you can use more horizontal space :-)

Also, I suggest the following change, more to the point:
```
Use the `embed_dim` from the config (stored in the class) rather than `hidden_state` because `attn_output` can be partitioned across GPUs when using tensor-parallelism.
```

Reply:
Sure, I will reformat this :)